packages/twenty-front/src/pages/onboarding/PaymentSuccess.tsx

```diff
@@ -36,16 +40,32 @@ const StyledButtonContainer = styled.div`
 export const PaymentSuccess = () => {
   const theme = useTheme();
   const currentUser = useRecoilValue(currentUserState);
+  const subscriptionStatus = useSubscriptionStatus();
+  const [getCurrentUser] = useGetCurrentUserLazyQuery();
+  const setCurrentUser = useSetRecoilState(currentUserState);
   const color =
     theme.name === 'light' ? theme.grayScale.gray90 : theme.grayScale.gray10;
 
+  const checkSubscriptionStatus = async () => {
+    if (!isDefined(subscriptionStatus)) {
+      const result = await getCurrentUser({ fetchPolicy: 'network-only' });
+      const currentUser = result.data?.currentUser;
+      if (isDefined(currentUser)) {
+        setCurrentUser(currentUser);
+      }
+    }
+  };
+
+  const onClick = async () => {
```

Comment:
does not look useful

if it was, we would name it handleButtonClick
onClick is the event and callbacks are always named in infinitive form